### ISSUE TITLE
Token retrieval must not delay editor initialization

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-cloudservices#9]
Right now retrieving the token block editor initialization which is bad for UX. In most cases, CS doesn't need to be up for the editor to be functional.

So, IMO, token can be retrieved later and situations when it's not yet retrieved should be handled in some graceful way.

**Comment by Reinmar** [comment]
cc @pjasiun @czerwonkabartosz

**Comment by czerwonkabartosz** [comment]
Yep, it's a good idea 👍

**Comment by pjasiun** [comment]
I agree it is a good idea. Note that now, it is as efficient as it was before when a client had to request token before editors initialization.

Not it is done in a similar way as connecting to the CS web socket API. However, that's true that the case is different. Usually, you need to have the connection to the web socket established when you init other plugins (like collaboration or comments), but plugins which use token for upload don't need it at the beginning. So, this plugin could request token asynchronously.

Since we are talking about the token request optimisation, there could be a separate configuration where you can set the initial token. Some server could generate it together with the HTML, and the additional request at the beginning won't be needed. It does not mean that the token URL is not needed, it would be still used then the token get invalid, but it could be used for the first authentication.
